### PR TITLE
Refactor CliTable to support logger

### PR DESCRIPTION
**Pull request by jstvz**
This PR updates `CliTable` to support the use of loggers where needed by adding a `table` property that returns a properly formatted string.

Also, updated `PackageUpload` to use `self.logger` for reporting apex test failures.

---
# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: cumulusci/cli/ui.py [hunk 1]
```diff
@@ -89,27 +89,29 @@ def echo(self, plain=False):
...
         if plain or os.environ.get("TERM") == "dumb":
-            self.ascii_table()
-            return None
+            table = self.ascii_table()
+        else:
+            table = self.table
+        click.echo(table)
 
+    @property
+    def table(self):
```
Comment (davisagli) [inline cumulusci/cli/ui.py]: What about `__str__`?

Comment (jstvz) [reply cumulusci/cli/ui.py]: Huh, forgot about `__str__`. Instead of this, or in addition to it?

Comment (davisagli) [reply cumulusci/cli/ui.py]: Instead of. I think `str(table)` is a clearer API than `table.table`

Comment (jstvz) [reply cumulusci/cli/ui.py]: Well, when you write it out like that... :smirk:

I'll update.

**Pull request merged**